packages/indexer/src/indexer.ts

```diff
@@ -34,8 +34,13 @@ export interface IndexerConfig<TFilter, TBlock, TRet> {
   finality?: DataFinality;
   startingCursor?: Cursor;
   factory?: (block: TBlock) => { filter?: TFilter; data?: TRet };
-  transform: (block: TBlock) => TRet;
-  sink?: Sink<TRet>;
+  transform: (args: {
+    block: TBlock;
+    cursor?: Cursor | undefined;
+    endCursor?: Cursor | undefined;
+    finality: DataFinality;
+  }) => TRet;
+  sink?: Promise<Sink<TRet>>;
```

Comment:
We should still allow users to pass a non-async function.

```suggestion
  sink?: Sink<TRet> | Promise<Sink<TRet>>;
```